.github/workflows/doc-build.yml

```diff
@@ -0,0 +1,61 @@
+name: Quarkus Documentation CI
+
+on:
+  push:
+    paths:
+      - 'docs/src/main/asciidoc/**'
+  pull_request:
+    paths:
+      - 'docs/src/main/asciidoc/**'
+
+
+jobs:
+  build-doc:
+    name: "Documentation Build"
+    runs-on: ubuntu-latest
+    # Skip draft PRs and those with WIP in the subject, rerun as soon as its removed
+    if: "github.event_name != 'pull_request' || ( \
+             github.event.pull_request.draft == false && \
+             github.event.pull_request.state != 'closed' && \
+             contains(github.event.pull_request.title, 'wip ') == false && \
+             contains(github.event.pull_request.title, '[wip]') == false && \
+             (
+               github.event.action != 'edited' || \
+               contains(github.event.changes.title.from, 'wip ') || \
+               contains(github.event.changes.title.from, '[wip]') \
+             ) \
+           )"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: n1hility/cancel-previous-runs@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up JDK 11
+        # Uses sha for added security since tags can be updated
+        uses: joschi/setup-jdk@b9cc6eabf7e7e3889766b5cee486f874c9e1bd2d
+        with:
+          java-version: 11
+      - name: Compute cache restore key
+        # Always recompute on a push so that the maven repo doesnt grow indefinitely with old versions
+        run: |
+          if ${{ github.event_name == 'pull_request' }}; then echo "::set-env name=COMPUTED_RESTORE_KEY::q2maven-"; fi
+      - name: Cache Maven Repository
+        id: cache-maven
+        uses: n1hility/cache@v2
+        with:
+          path: ~/.m2/repository
+          # Improves the reusability of the cache to limit key changes
+          key: q2maven-${{ hashFiles('bom/runtime/pom.xml') }}
+          restore-keys: ${{ env.COMPUTED_RESTORE_KEY }}
+          restore-only: ${{ github.event_name == 'pull_request' }}
+      - name: Build
+        run: |
+          mvn -e -B --settings .github/doc-mvn-settings.xml clean package -pl docs
+      - name: Tar Maven Repo
+        shell: bash
+        run: tar -czf maven-repo.tgz -C ~ .m2/repository
+      - name: Persist Maven Repo
+        uses: actions/upload-artifact@v1
+        with:
+          name: maven-repo
+          path: maven-repo.tgz
```

Comment:
I wonder if we should avoid storing the cache, given there's a good chance it won't bring anything useful.

Reply:
Without cache, the downloading of all the JARs needed for the plugins took multiple minutes

Reply:
I'm not saying to not use the cache. I'm saying to not push it. We can reuse the one we usually have around but not push it.

Also I think just executing the Asciidoctor plugin might help.

Reply:
Also, we definitely don't want to pollute the cache with snapshots.

Reply:
Removed the upload of the cache